sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -608,6 +608,34 @@ class FileStreamSourceSuite extends FileStreamSourceTest {
 
   // =============== other tests ================
 
+  test("read new files in partitioned table without globbing, should read partition data") {
```

Comment:
We should probably also have an explicit test for the case where schema inference is turned on (you implicitly test it some with the code changed below)